src/cpp/session/modules/quarto/SessionQuarto.cpp

```diff
@@ -1207,9 +1210,56 @@ void readQuartoProjectConfig(const FilePath& configFile,
    }
 }
 
+// Given the (aliased) path to a file, return the file path to the working directory where code
+// should be executed in the file, based on the settings in _quarto.yml.
+//
+// Returns an empty FilePath if no directory is specified.
+FilePath getQuartoExecutionDir(const std::string& docPath)
+{
+   // Ensure we have a path to work with (an empty string will resolve to the home directory below)
+   if (docPath.empty())
+   {
+      return FilePath();
+   }
 
+   // Resolve path, and ensure the document is a Quarto file
```

Comment:
From what I understand, R Markdown documents within a Quarto project might be treated like Quarto documents (not sure if it depends on the contents of the YAML header or not, though). Does that need to be handled here?

Reply:
This should be resolved now that Quarto detection has been offloaded to recycle the logic in `onDetectQuartoSourceType`.